modules/custom/openy_home_branch/js/hb_cookies_storage.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ * Cookies storage JavaScript for the Open Y Home Branch module.
+ */
+
+(function ($, Drupal, drupalSettings) {
+
+  "use strict";
+
+  Drupal.homeBranch = {
+
+    // Home branch data.
+    data: {
+      id: null,
+      dontAsk: false
+    },
+
+    // Locations list.
+    locations: {},
+
+    // Plugins that can be provided by other components.
+    // @see openy_home_branch/js/hb-plugins/hb-location-finder.js
+    // Drupal.homeBranch.plugins.push.
+    plugins: [],
+
+    // Object Initialization.
+    init: function () {
+      this.data = this.loadFromStorage();
+
+      // Get locations list.
+      let self = this;
+      self.locations = drupalSettings.home_branch.hb_menu_selector.locations;
+
+      setTimeout(function() {
+        self.attachPlugins();
+      }, 0);
+    },
+
+    // Attaches plugins.
+    attachPlugins: function () {
+      if (this.plugins.length === 0) {
+        return;
+      }
+      Drupal.homeBranch.plugins.forEach(function (plugin, key, arr) {
+        plugin.attach(plugin.settings);
+      });
+      if (this.data['id'] == null && !this.data['dontAsk']) {
+        this.showModal();
+      }
+    },
+
+    // Get property value.
+    getValue: function (property) {
+      return this.data[property];
+    },
+
+    // Set values.
+    set: function (id, dontAsk) {
+      this.data.id = id;
+      this.data.dontAsk = dontAsk;
+      this.updateStorage();
+    },
+
+    // Set property value.
+    setValue: function (property, value) {
+      this.data[property] = value;
+      this.updateStorage();
+    },
+
+    // Set property value.
+    setId: function (id) {
+      this.setValue('id', id);
+    },
+
+    // Get locations list.
+    getLocations: function () {
+      return this.locations;
+    },
+
+    // Move current data to Cookies storage.
+    updateStorage: function () {
+      $(document).trigger('hb-before-storage-update', this.data);
+      $.cookie('home_branch', JSON.stringify(this.data), { expires: 360, path: '/' });
```

Comment:
can we avoid of setting path? @hamrant

Reply:
no, without path it can set a different cookie on each page with the HB plugin

Reply:
Not specifying path means setting the cookie for the current page only == HB just won't work as the cookie is supposed to work across the site.

Setting it to `/` means setting to the whole domain == doesn't work properly for these builds.

Setting it to the basepath should resolve it. In other words, the base path `/buildXXX/` should be passed to the storage for the builds, and `/` for basic websites.

@hamrant @podarok Will it work for you?

Reply:
@AndreyMaximov yes, we can use `drupalSettings.path.baseUrl`